src/matcha_ml/storage/azure_storage.py

```diff
@@ -83,6 +85,10 @@ def upload_folder(self, container_name: str, src_folder_path: str) -> None:
             for filename in filenames:
                 file_path = os.path.join(root, filename)
 
+                # ignore uploading files in IGNORE_FOLDERS
+                if any(ignore_folder in file_path for ignore_folder in IGNORE_FOLDERS):
```

Comment:
Would this be better as a set operation and as a one-liner?

```python
IGNORE_FOLDERS = {'.terraform'}

if (file_path not in IGNORE_FOLDERS) and (file_path in blob_set):
    blob_set.remove(file_path)
```

Reply:
Will create a separate ticket as improvement for this.